SCPlayer/SCPlayer/Controllers/Cores/Home Screen/HomeViewController.swift

```diff
@@ -62,6 +62,23 @@ final class HomeViewController: UIViewController {
             self.homeCollectionView.reloadData()
         }
     }
+    
+    private func loadLikedStatusData(tracks: [Track]) -> [Track] {
+        var listTrack = tracks
+        guard let listLikedTrackId = TrackDatabase.shared.queryAll() else {
+            return []
+        }
+        var count = 0
+        while count < listTrack.count {
+            if listLikedTrackId.contains(tracks[count].trackID ?? 0) {
+                listTrack[count].isLiked = true
+            } else {
+                listTrack[count].isLiked = false
+            }
+            count += 1
+        }
+        return listTrack
```

Comment:
```suggestion
        guard let listLikedTrackId = TrackDatabase.shared.queryAll() else {
            return []
        }
        let listTrack = tracks.map { $0.isLiked = listLikedTrackId.contains($0.trackID ?? 0) }
        return listTrack
```
Em thử cái này xem ok không

Reply:
Anh ơi sử dụng Map thế này nó sẽ trả về listTrack là 1 mảng Bool của thuộc tính isLiked. Em nghĩ một hồi và code thử nhiều kiểu từ map, fillter nhưng em thấy không triển khai được. Vậy em xử dụng vòng lặp while này được không anh ?